### PR TITLE
[BUG] import dict metric data in SMAC and BOHB

### PR DESCRIPTION
**Pull request by QuanluZhang**

### CODE REVIEW

File: src/sdk/pynni/tests/test_builtin_tuners.py [hunk 1]
```diff
@@ -141,36 +141,94 @@ def search_space_test_all(self, tuner_factory, supported_types=None, ignore_type
...
                                    ignore_types=["uniform_equal", "qloguniform_equal", "loguniform_equal", "quniform_clip_2"])
         # NOTE: types are ignored because `tpe.py line 465, in adaptive_parzen_normal assert prior_sigma > 0`
+        self.import_data_test(lambda: HyperoptTuner("tpe"))
 
     def test_random_search(self):
         self.search_space_test_all(lambda: HyperoptTuner("random_search"))
+        self.import_data_test(lambda: HyperoptTuner("random_search"))
 
     def test_anneal(self):
         self.search_space_test_all(lambda: HyperoptTuner("anneal"))
```
Comment (ultmaster) [inline src/sdk/pynni/tests/test_builtin_tuners.py]: I would write:

```python
tuner_fn = lambda: HyperoptTuner("anneal")
self.search_space_test_all(tuner_fn)
self.import_data_test(tuner_fn)
```

It's better-looking.

Comment (QuanluZhang) [reply src/sdk/pynni/tests/test_builtin_tuners.py]: updated

File: src/sdk/pynni/tests/test_builtin_tuners.py [hunk 2]
```diff
@@ -141,50 +141,119 @@ def search_space_test_all(self, tuner_factory, supported_types=None, ignore_type
             logger.info("Full supported search space: %s", full_supported_search_space)
             self.search_space_test_one(tuner_factory, full_supported_search_space)
 
+    def import_data_test(self, tuner_factory, stype="choice_str"):
+        # import data at the beginning with number value and dict value
```
Comment (xuehui1991) [inline src/sdk/pynni/tests/test_builtin_tuners.py]: use doc string """ """ here.

Comment (QuanluZhang) [reply src/sdk/pynni/tests/test_builtin_tuners.py]: updated

**Pull request merged**